### DISCUSSION TITLE
Auto login or disable login?

### DISCUSSION THREAD

**Discussion by cysec404** [mealie-recipes/mealie#284] [Q&A]
Hi, I'm really enjoying this app, it's super user friendly and easy to use so I think my family will finally adopt it as a central repository for all our recipes. 

Is there a way to not require a login and retain the same functionality of a logged in user?  This is only accessible in my home network. For my family to use and adopt it like I want them to, I need to remove the need to login. I would like any of them to be able to pull up the page and start using it, whether just reading a recipe or entering a new one.

Thanks so much!

**Comment by hay-kot** [comment]
Currently there is no plan for this. Managing some sort of switch to allow users to turn on/off login is too much for me to maintain long term. In the future, I'll be adding an environmental variable to set token timeouts. Then you could have your family login once and then set an arbitrary long token time so they only have to re-login every once in a while.

**Reply by jsievenpiper** [reply]
Is there room for an option for providing authentication via HTTP headers or otherwise? I love mealie, and have it running behind SSO with 2FA, so having a third auth method in this flow isn't just overkill but also means there's another set of users to manage that are application-specific. It seems that _most_ applications intended to be self-hosted offer some sort of functionality like this.

**Reply by hay-kot** [reply]
> Is there room for an option for providing authentication via HTTP headers or otherwise? I love mealie, and have it running behind SSO

For sure! I'd love to see the functionality come to Mealie. I think that different than #293 that you posted on. That said...  I'm not going to be working on this any time soon, It's not something I would use, and I think there's a lot of core functionality that takes priority over SSO that I need to focus on. I don't think it would be crazy difficult to get this working. We're already using OAuth2 so I think it would just be a matter of a few ENV variables and some route dependency swapping. 

If anyone wants to take this project on, I'd love to see it and help out where I can.

**Reply by Christoph-87** [reply]
I do have the same issue and would love to take this forward.

**Reply by hay-kot** [reply]
> I do have the same issue and would love to take this forward.

Which part? There is a TOKEN_TIME variable that you can set. [See Here](https://hay-kot.github.io/mealie/documentation/getting-started/install/#env-variables). If you're talking about SSO, 2FA, or Basic Auth, best to open a Feature Request and we can track that there. I'm open to support on those, it isn't something I'm actively looking at.

**Reply by Christoph-87** [reply]
I am using Authelia for all my applications to serve the authentification. When using Mealie, I have to login to Authelia and then again to Mealie as it does not support authentification via HTTP headers yet. Using Authelia can give me the benefit of having 2FA and it will log me in to other applications that support authentification via HTTP headers, too. 
Here is an example of Organizr, They implemented it with custom headers that can be used to provide the username and mail. In addition a trusted proxy can be given: https://github.com/causefx/Organizr/issues/1215

**Reply by Northguy** [reply]
@Christoph-87 , I am also using Authelia as a forward auth on my reverse proxy. I think best course of action for a lot of projects would be to abandon the authentication via Http headers  but switch to OpenID Connect. I have this configured in combination with Bookstack and this works like a charm

**Reply by Christoph-87** [reply]
@Northguy Yes, I also think that OpenID Connect would be the best. But I will not be able to contribute anything for this, because I do not know exactly how this works. Does OpenID Connect already work with Authelia since I see from the documentation that it is still in beta? https://www.authelia.com/docs/configuration/identity-providers/oidc.html

**Reply by Northguy** [reply]
@Christoph-87 yes, it is functional but not out of Beta stage yet. As you can see in that table, there are still enhancements to be made. For non-production use in a home environment it is perfectly useable though. And as said: I have it working with Bookstack and planning to integrate it with some other apps in near future.

**Reply by tekchip** [reply]
Would GRC's SQRL be a solution here? https://www.grc.com/sqrl/sqrl.htm If i'm missing the goal or the point here I apologize.

**Reply by BobWs** [reply]
How to use / setup OAuth2 ? I can't find any documentation about it.
Love your app by the way, testing it at the moment and compering it with Tandoor/Recipes. But at first sight I do prefer your way of setup and ease...

**Reply by hay-kot** [reply]
@BobWs OAuth2 isn't currently supported.

Not something that's on my radar, but happy to accept a Pull Request that provides this functionality.

**Comment by virtux** [comment]
Hi @hay-kot, in 2024 is it possible disable login or a workaround (token-time, define safe network, autologin...)
For home users, it would be helpful to allow the family to add recipes without remembering and logging in each time.

Thanks!
ps great job with the software!

**Reply by boc-the-git** [reply]
@virtux in 2024 security is more important than ever!

There is an environment variable `TOKEN_TIME` to set the number of hours a session lasts for. Give that a go, and see if it works for you 😄 
https://nightly.mealie.io/documentation/getting-started/installation/backend-config/

**Comment by Prekzz** [comment]
On the same note as the `TOKEN_TIME` parameter. Is it possible to just completely run it internally, especially without the use of HTTPS/SSL. And still can copy things from my shopping list?

**Reply by hay-kot** [reply]
No, this is a browser limitation and not something we can do anything about.

**Comment by dimxyp** [comment]
I honestly don’t understand why this feature isn’t enabled in the app...
What are we afraid of — that someone will steal your pie recipe?
I have 3 people using it for family purposes, and no matter what I do with the token_time, it still asks for a password.
And since everyone gets annoyed, no one ends up using it at all...
I really don’t see the point of authentication in this kind of app, especially when you can run it only on your internal network.

**Reply by hay-kot** [reply]
We didn't use to have auth. The number one request from users was auth. You're in the minority here.

If token time isn't working it's possible your browser is configured to delete data and cookies when it closes. I would check that.

**Reply by dimxyp** [reply]
Okay, I understand that some might want authentication, perhaps using it in professional environments, as the code is remarkably valuable and might be integrated into their work. (and maybe they expose mealie to internet)  However, it would be nice to have an option in the settings to enable, for example, home-based use and be a tool for the whole family without login page.

Just so you know, I have a tablet with Home Assistant with tab always open with Mealie in the kitchen, with over 150 recipes. I constantly get complains that it asks for a password, and they can't view the recipes. 
after a while they just open old recipe book instead :)

**Reply by michael-genson** [reply]
> I constantly get complains that it asks for a password

If you have to sign in more than once, something is configured wrong on your end. That's what the TOKEN_TIME variable is for.

Also, if you're using it for reading recipes only, you can configure Mealie to be browsable without a login. Login is only required for user actions

**Reply by dimxyp** [reply]
whouou I didn't know that!! I will configure that.. i dont need modification of my data most of the time

thanks a lot @michael-genson

**Reply by michael-genson** [reply]
No problem! More info in the docs: https://docs.mealie.io/documentation/getting-started/usage/permissions-and-public-access/

**Reply by Robubble** [reply]
> If you have to sign in more than once, something is configured wrong on your end. That's what the TOKEN_TIME variable is for.

Why is token time needed at all when there is a "stay logged in/remember me" checkbox?
